core/block_validator.go

```diff
@@ -30,23 +30,38 @@ import (
 
 const badBlockCacheExpire = 30 * time.Second
 
+type BlockValidatorOption func(*BlockValidator) *BlockValidator
+
+func EnableRemoteVerifyManager(remoteValidator *remoteVerifyManager) BlockValidatorOption {
+	return func(bv *BlockValidator) *BlockValidator {
+		bv.remoteValidator = remoteValidator
+		return bv
```

Comment:
Is it a bit weird to return `bv`, since it is passed in as self pointer?

Reply:
I just want to keep the same style as the `BlockChainOption` function